homeassistant/components/nut/__init__.py

```diff
@@ -131,7 +132,7 @@ def _firmware_from_status(status):
 def _serial_from_status(status):
     """Find the best serialvalue from the status."""
     serial = status.get("device.serial") or status.get("ups.serial")
-    if serial and serial == "unknown":
+    if serial and (serial.lower() == "unknown" or re.search(r"^0+$", serial)):
```

Comment:
Alternative:

```suggestion
    if serial and (serial.lower() == "unknown" or serial.count("0") == len(serial)):
```

Reply:
that seems more pythonic, lets go with that.  I'm so used to writing out a 200 char regex.